src/csharp/Grpc.Core/Internal/PlatformApis.cs

```diff
@@ -33,12 +33,17 @@ namespace Grpc.Core.Internal
     internal static class PlatformApis
     {
         const string UnityEngineApplicationClassName = "UnityEngine.Application, UnityEngine";
+        const string XamarinAndroidActivityClassName = "Android.App.Activity, Mono.Android";
+        const string XamariniOSEnumClassName = "Mono.CSharp.Enum, Mono.CSharp";
```

Comment:
is presence of "Mono.CSharp.Enum, Mono.CSharp" really an indicator we're running on iOS?

Reply:
No, this is plain wrong. Since we only had a look at Android, I had a quick guess at what type would be suitable to query for on monotouch.

This should probably be changed to `MonoTouch.Foundation.NSObject, monotouch` (haven't tested it)

Reply:
I think I'll go with what's suggested in https://github.com/grpc/grpc/pull/16089/files#diff-febca96a7ab787e84fd44d7c53f09a44R37 (and will fix that in a followup PR)